### PR TITLE
chore(electric): Cleanup satellite- and postgres-connector-related code

### PR DESCRIPTION
**Pull request by alco**
Following on from https://github.com/electric-sql/electric/pull/1044, these are a bunch of refactorings that didn't logically fit in any other PR.

### CODE REVIEW

**Review by magnetised — APPROVED** [review]
Fab. Putting the origin into the protocol state has been on my list for a while.

File: components/electric/lib/electric/replication/postgres_connector_sup.ex [hunk 1]
```diff
@@ -21,44 +23,39 @@ defmodule Electric.Replication.PostgresConnectorSup do
...
-      %{
-        id: :postgres_producer,
-        start: {Postgres.LogicalReplicationProducer, :start_link, [connector_config]}
-      },
-      %{
-        id: :postgres_migration_consumer,
+        id: Postgres.MigrationConsumer,
         start:
           {Postgres.MigrationConsumer, :start_link, [connector_config, migration_consumer_opts]}
       },
```
Comment (icehaunter) [inline components/electric/lib/electric/replication/postgres_connector_sup.ex:50]: Nitpick: if you're removing ids here, why not do this too?

```suggestion
{Postgres.MigrationConsumer, {connector_config, migration_consumer_opt}}
```

Comment (icehaunter) [reply components/electric/lib/electric/replication/postgres_connector_sup.ex:50]: sorry, the suggestion built up incorrectly, it's just over the migration consumer

**Review by icehaunter — APPROVED** [review]
Just a small nitpick, LGTM otherwise!

File: components/electric/lib/electric/replication/postgres_connector_sup.ex (hunk 1 above)
Comment (alco) [reply components/electric/lib/electric/replication/postgres_connector_sup.ex:50]: Perfect! I overlooked the tuple approach myself.

**Comment by icehaunter** [comment]
Sorry @alco! I missed that this merge didn't target `main`.

**Comment by alco** [comment]
I have recreated this PR as https://github.com/electric-sql/electric/pull/1144.

**Pull request merged**